splits/readers.py

```diff
@@ -57,8 +57,10 @@ def read(self, num=None):
 
                 if num > 0 and len(val) == num:
                     break
-        except:
+        except StopIteration:
             pass
+        except Exception as e:
+            raise Exception("Unhandled exception in SplitReader:"+repr(e))
```

Comment:
I think you should find the specific exception that happens for data corruption and raise, that will be safer than raising every single exception.